### PR TITLE
BFDProfile: Fix: Deleted profile not being updated in speakers

### PR DESCRIPTION
**Pull request by sabinaaledort**
If a BFD profile was deleted and there are no more configured BFD profiles (the length of profiles is 0) the sync function should be called to update the config and not skipped.

### CODE REVIEW

**Comment by cgoncalves** [comment]
Thanks for the fix!
Would it be possible to add test coverage? E2E would be great, if possible.

**Comment by fedepaol** [comment]
> Thanks for the fix! Would it be possible to add test coverage? E2E would be great, if possible.

+1 , if unit tests are too narrow to cover this, another good place is to extend this test https://github.com/metallb/metallb/blob/31080d6f4e06f7d579127034b8ecc28e5ca702d4/e2etest/bgptests/bgp.go#L1076 by removing the profile and inspecting the file again.

File: e2etest/bgptests/bgp.go [hunk 1]
```diff
@@ -1136,6 +1136,35 @@ var _ = ginkgo.Describe("BGP", func() {
...
+					cfgStr, err := podExecutor.Exec("vtysh", "-c", "show running-config")
+					if err != nil {
+						return err.Error()
+					}
+
+					return cfgStr
+				}, 1*time.Minute).Should(
+					And(
+						ContainSubstring("log file /etc/frr/frr.log informational"),
+						WithTransform(substringCount("\n profile fullbfdprofile1"), Equal(0)),
```
Comment (cgoncalves) [inline e2etest/bgptests/bgp.go:1164]: Can we also validate that the BGP peer does not have BFD enabled with the default BFD profile?

This is, should not have `neighbor $NEIGHBOR_IP bfd`

**Review by cgoncalves — CHANGES_REQUESTED** [review]

File: e2etest/bgptests/bgp.go [hunk 2]
```diff
@@ -1136,6 +1136,43 @@ var _ = ginkgo.Describe("BGP", func() {
 					),
 				)
 			}
+
+			ginkgo.By("validating speakers running-config is updated after deleting the BFD profile")
```
Comment (pperiyasamy) [inline e2etest/bgptests/bgp.go:1140]: should we also validate if peers are referring to an existing BFD profile ?

File: e2etest/bgptests/bgp.go (hunk 1 above)
Comment (sabinaaledort) [reply e2etest/bgptests/bgp.go:1164]: I added the test.
It seems we found a bug in FRR. The BFD profile is removed from the FRR config but the BGP peers still have BFD enabled with: `neighbor $NEIGHBOR_IP bfd`.

Comment (fedepaol) [reply e2etest/bgptests/bgp.go:1164]: Can you check that the renderer file does not contain them (i.e. it's an frr bug and not a bug on our side)?

File: e2etest/bgptests/bgp.go (hunk 2 above)
Comment (sabinaaledort) [reply e2etest/bgptests/bgp.go:1140]: if the BFD profile doesn't exist we produce an error [here](https://github.com/metallb/metallb/blob/81a0de8f7a8e2d073590d77e55065f14df97d698/internal/config/config.go#L282) which is printed to the speakers log, i think checking the speakers log can be a bit tricky.

File: e2etest/bgptests/bgp.go (hunk 1 above)
Comment (sabinaaledort) [reply e2etest/bgptests/bgp.go:1164]: I checked the rendered file, it doesn't have BFD configuration at all.
We tested it manually as well and got the same result.

File: e2etest/bgptests/bgp.go (hunk 2 above)
Comment (pperiyasamy) [reply e2etest/bgptests/bgp.go:1140]: oh! no, both states are just used to remove the entry from queue, looks this test is not feasible.

Comment (pperiyasamy) [reply e2etest/bgptests/bgp.go:1140]: we may use `metallb_k8s_client_update_errors_total` metric if that throws `SyncStateError`, but i'm not sure if this PR intended to do this.

Comment (liornoy) [reply e2etest/bgptests/bgp.go:1140]: @sabinaaledort If you are interested in checking speakers logs you can use what I wrote on my PR
https://github.com/metallb/metallb/pull/1177/files#diff-dcd1cc413d712c08cb1ff01c2fb08dcaa495e4d2ed52e75d2367fc6b6da7ca82

**Comment by fedepaol** [comment]
@sabinaaledort this is on hold until frr is fixed right?

**Comment by sabinaaledort** [comment]
> @sabinaaledort this is on hold until frr is fixed right?

yes.

**Pull request closed without merging**